### PR TITLE
XML extensions to mirror class annotations trace lambda and trace by return type behaviour

### PR DESCRIPTION
**Pull request by GDownes**
### Overview
Added two new XML properties to provide the ability to apply logic similar to the new class annotations TraceLambda and TraceByReturnType using XML.

### Testing
Added unit tests to cover XML TraceLambda TraceByReturnType usage scenarios.
Deploy updated agent locally and reviewed traces in New Relic One.

### Checks

[X] Are your contributions backwards compatible with relevant frameworks and APIs? Yes
[X] Does your code contain any breaking changes? Please describe. No
[X] Does your code introduce any new dependencies? Please describe. No

### CODE REVIEW

**Review by meiao — CHANGES_REQUESTED** [review]

File: newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/methodmatchers/LambdaMethodMatcher.java [hunk 1]
```diff
@@ -29,7 +29,7 @@ public LambdaMethodMatcher(String pattern, boolean includeNonstatic) {
 
     @Override
     public boolean matches(int access, String name, String desc, Set<String> annotations) {
-        return (includeNonstatic || access == Opcodes.F_NEW) && lambdaMethodPattern.matcher(name).matches();
+        return (includeNonstatic || access == 4121 || access == Opcodes.F_NEW) && lambdaMethodPattern.matcher(name).matches();
```
[the remark this answers is not in the document]
Comment (GDownes) [reply newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/methodmatchers/LambdaMethodMatcher.java:32]: Added constant ACCESS_SYNTHETIC_STATIC to class for int 4121

File: newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/methodmatchers/LambdaMethodMatcher.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * A method matcher that matches lambda methods.
+ */
+public final class LambdaMethodMatcher implements MethodMatcher {
+
+    private final Pattern lambdaMethodPattern;
+    private final Boolean includeNonstatic;
```
[the remark this answers is not in the document]
Comment (GDownes) [reply newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/methodmatchers/LambdaMethodMatcher.java:22]: Updated to boolean

File: newrelic-agent/src/main/java/com/newrelic/agent/extension/beans/Extension.java [hunk 3]
```diff
@@ -643,6 +647,85 @@ public void setIgnoreTransaction(Boolean value) {
...
+            public Boolean getIncludeNonstatic() {
+                return includeNonstatic;
+            }
+
+            /**
+             * Sets the value of the includeNonstatic property.
+             *
+             * @param value allowed object is {@link Boolean }
+             */
+            public void setIncludeNonstatic(Boolean value) {
```
[the remark this answers is not in the document]
Comment (GDownes) [reply newrelic-agent/src/main/java/com/newrelic/agent/extension/beans/Extension.java:725]: Updated to boolean

**Review by meiao — APPROVED** [review]

**Comment by meiao** [comment]
AITs green on https://javaagent-build.pdx.vm.datanerd.us/job/JavaAgent_PR_AIT/96/

**Pull request merged**